staging/kos/cmd/controllers/connectionagent/main.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"os"
+	"os/signal"
+	"runtime"
+	"syscall"
+	"time"
+
+	"github.com/golang/glog"
+
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/workqueue"
+
+	kosclientset "k8s.io/examples/staging/kos/pkg/client/clientset/versioned"
+	cactlr "k8s.io/examples/staging/kos/pkg/controllers/connectionagent"
+	netfabricfactory "k8s.io/examples/staging/kos/pkg/networkfabric/factory"
+)
+
+const (
```

Comment:
Why use environment variables?  Explicit command line flags are clearer.

Reply:
The node name and the host IP are initialized with environment variables, with the instructions `localNodeName, localNodeEnvSet := os.LookupEnv(nodeNameEnv)` and `hostIP, hostIPEnvSet := os.LookupEnv(hostIPEnv)` respectively because then it's very convenient to set these vars in the yaml manifest for the DaemonSet of the connection agent:https://github.com/MikeSpreitzer/kube-examples/blob/2bdca7a2945c8ea6466b8a64e1c71b35c3292255/staging/kos/deploy.m4/50-ds-ca.yaml.m4#L17-L34

Specifically, the fact that we can use a fieldPath in the yaml to specify the value of the environment variable makes it impossible for the aforementioned variables to take incorrect values. I used env variables because I was already familiar with using a fieldPath to initialize them, and was not aware that you could do something similar for the container invocation arguments. I agree that invocation arguments are more explicit, I'll add them (I think we can fall back to env vars if they are not set).

Reply:
Added command line arguments in d0dce6b

Reply:
Yes, it would be nice to have a more direct path.  That is something to mention to the upstream community.

Now that we are using command line flags, let us not _also_ bother with environment variables in the code, just for the sake of simplicity.

Reply:
Removed env vars in 06d37b6. I will investigate and open a feature request for container args referencing directly pod fields.